vtexdocs/devportal
# build(rapidoc): add script to fetch rapidoc-min.js from vtexdocs/RapiDoc

#### What is the purpose of this pull request?

To add a script capable of fetching and writing the RapiDoc build file, from the vtexdocs/RapiDoc repository, into the devportal repo.

#### What problem is this solving?

Currently the RapiDoc build is being accessed with a CDN. To achieve better styling on OpenAPI rendering, we now have our own [RapiDoc fork](https://github.com/vtexdocs/RapiDoc), from where we should get the file `dist/rapidoc-min.js` to be used at the devportal.
This PR proposes a script that fetches the file according to the rapidoc version manually defined in the `package.json` and saves it in the path` public/rapidoc/rapidoc-min.js` during the pre-build and pre-dev phases of the app. That file is already being properly referenced for OpenAPI rendering, and the initial version (tag 1.0.0) has already been defined both in the vtexdocs/RapiDoc repository and in devportal/package.json.

#### How should this be manually tested?

As the script should already be running in the pre-build and pre-dev stages, you should [access this deploy preview](https://deploy-preview-36--elated-hoover-5c29bf.netlify.app/) and try to access the API reference links (as the Catalog.json). Check if the OpenAPI is being properly rendered by comparing with [the official current deploy here](https://elated-hoover-5c29bf.netlify.app/).

[Further information about this task should be found here (focus on the third described approach).](https://www.notion.so/vtexhandbook/Cria-o-e-configura-o-do-repo-Rapi-Doc-na-nova-organiza-o-vtexdocs-3f507915435e4ed3a4b167af50d86685)
